python/ribasim/ribasim/model.py

```diff
@@ -1,43 +1,84 @@
 import datetime
 import inspect
 import shutil
-from contextlib import closing
 from pathlib import Path
-from sqlite3 import connect
-from typing import Any, Optional, Type, cast
+from typing import Any, Dict, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 import tomli
 import tomli_w
-from pydantic import BaseModel
-
-from ribasim import geometry, node_types
-from ribasim.config import Allocation, Logging, Solver
-from ribasim.geometry.edge import Edge
-from ribasim.geometry.node import Node
+from pydantic import DirectoryPath, model_serializer, model_validator
+
+from ribasim import node_types
+from ribasim.config import (
+    Allocation,
+    Basin,
+    DiscreteControl,
+    FlowBoundary,
+    FractionalFlow,
+    LevelBoundary,
+    LinearResistance,
+    Logging,
+    ManningResistance,
+    Outlet,
+    PidControl,
+    Pump,
+    Results,
+    Solver,
+    TabulatedRatingCurve,
+    Terminal,
+    User,
+)
+from ribasim.geometry.edge import Edge, EdgeStaticSchema
+from ribasim.geometry.node import Node, NodeStaticSchema
 
 # Do not import from ribasim namespace: will create import errors.
 # E.g. not: from ribasim import Basin
-from ribasim.input_base import TableModel
-from ribasim.node_types.basin import Basin
-from ribasim.node_types.discrete_control import DiscreteControl
-from ribasim.node_types.flow_boundary import FlowBoundary
-from ribasim.node_types.fractional_flow import FractionalFlow
-from ribasim.node_types.level_boundary import LevelBoundary
-from ribasim.node_types.linear_resistance import LinearResistance
-from ribasim.node_types.manning_resistance import ManningResistance
-from ribasim.node_types.outlet import Outlet
-from ribasim.node_types.pid_control import PidControl
-from ribasim.node_types.pump import Pump
-from ribasim.node_types.tabulated_rating_curve import TabulatedRatingCurve
-from ribasim.node_types.terminal import Terminal
-from ribasim.node_types.user import User
+from ribasim.input_base import (
+    FileModel,
+    NodeModel,
+    TableModel,
+    context_file_loading,
+)
 from ribasim.types import FilePath
 
 
-class Model(BaseModel):
+class Database(NodeModel, FileModel):
```

Comment:
I couldn't find anything whether inheriting from multiple pydantic models is supported.
Might be good to double check: https://docs.pydantic.dev/latest/help_with_pydantic/